### ISSUE TITLE
UAA leaks connections to stats-injector

### ISSUE THREAD

**Issue by andrew-edgar** [cloudfoundry/uaa-release#216]
we have deployed cf-deployment 13.5 on our test env.  after several days bosh-dns fails and cannot come up again.  the bosh team said this may be because too many connections are established which makes it not possible for anything to connect to bosh-dns anymore.   doing a netstat -alnp we find that there are 28232 connections established that looks like this …

```
udp        0      0 127.0.0.1:47604         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:59892         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:51700         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:43508         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:39412         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:55797         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:51701         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:59893         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:47605         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:43509         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:39413         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:35317         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:59894         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:43510         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:47606         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:35318         127.0.0.1:8125          ESTABLISHED 938/java
udp        0      0 127.0.0.1:39414         127.0.0.1:8125          ESTABLISHED 938/java
```

pid 938 is
```
vcap       938   926 32 Jul06 ?        15:08:13 /var/vcap/packages/uaa/jdk/bin/java -Djava.util.logging.config.file=/var/vcap/data/uaa/tomcat/conf/logging.properties -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager -Djava.security.egd=file:/dev/./urandom -DPID=7 -Dnetworkaddress.cache.ttl=0 -Djavax.net.ssl.trustStore=/var/vcap/data/uaa/cert-cache/cacerts -Djavax.net.ssl.trustStoreType=JKS -Djavax.net.ssl.trustStorePassword=changeit -Dlog4j.configurationFile=/var/vcap/jobs/uaa/config/log4j2.properties -Djdk.tls.ephemeralDHKeySize=2048 -Djava.protocol.handler.pkgs=org.apache.catalina.webresources -Dorg.apache.catalina.security.SecurityListener.UMASK=0027 -Xmx768m -XX:MaxPermSize=256m -Dignore.endorsed.dirs= -classpath /var/vcap/data/uaa/tomcat/bin/bootstrap.jar:/var/vcap/data/uaa/tomcat/bin/tomcat-juli.jar -Dcatalina.base=/var/vcap/data/uaa/tomcat -Dcatalina.home=/var/vcap/data/uaa/tomcat -Djava.io.tmpdir=/var/vcap/data/uaa/tomcat/temp org.apache.catalina.startup.Bootstrap start
```

8125 is “statsd” injector.  seems like UAA is leaking connections to that

on the older version of UAA (74.18.0) vs (74.21.0)  and statsd-injector is (1.11.15) on both

on restart of UAA it quickly builds up more and more connections.

This problem eventually causes bosh-dns to fail and causes the the VM to be marked as failing.  This was originally reported as an issue here (https://github.com/cloudfoundry/bosh-dns-release/issues/65) but that can be closed as that is just a symptom of this UAA issue.

This problem would mean that 74.21 is bad and should be reported in the cf-deployment that consumes that.

**Comment by shamus** [comment]
Thanks for the report! 

Recently, some work was done to allow the UAA to emit metrics to statsd directly, instead of relying on the separate statsd application. The BOSH release should not be opting into this new configuration, but rather still rely on the `statsd.war`.

I'll speak with the @cf-uaa team today about finding the cause.

**Comment by thausler786** [comment]
This should be fixed as of [v74.23.0](https://github.com/cloudfoundry/uaa-release/releases/tag/v74.23.0)
